featuretools/tests/computational_backend/test_calculate_feature_matrix.py

```diff
@@ -1868,3 +1870,34 @@ def test_cfm_with_invalid_time_index(es):
     match += "which differs from other entityset time indexes"
     with pytest.raises(TypeError, match=match):
         calculate_feature_matrix(features=features, entityset=es)
+
+
+def test_cfm_with_lag_and_non_nullable_column(pd_es):
+    # fill nans so we can use non nullable numeric logical type in the EntitySet
+    new_log = pd_es['log'].copy()
+    new_log['value'] = new_log['value'].fillna(0)
+    new_log.ww.init(logical_types={'value': 'Integer',
+                                   "product_id": "Categorical"},
+                    index='id', time_index='datetime',
+                    name='new_log')
+    pd_es.add_dataframe(new_log)
+    rels = [('sessions', 'id', 'new_log', 'session_id'),
+            ('products', 'id', 'new_log', 'product_id')]
+    pd_es = pd_es.add_relationships(rels)
+
+    assert isinstance(pd_es['new_log'].ww.logical_types['value'], Integer)
+
+    lag_primitive = NumericLag(periods=5)
+    features = ft.dfs(target_dataframe_name='new_log',
+                      entityset=pd_es,
+                      agg_primitives=[],
+                      trans_primitives=[lag_primitive],
+                      features_only=True)
+
+    fm = calculate_feature_matrix(features=features, entityset=pd_es)
```

Comment:
Could remove `feautres_only=True) and combine these into a single call:
```
    fm, features = ft.dfs(target_dataframe_name='new_log',
                          entityset=pd_es,
                          agg_primitives=[],
                          trans_primitives=[lag_primitive],)
```

Also, should we do any checks on the features values or logical types? Not sure it's critical, but you could check that we end up with null values as expected. If we don't check anything in the feature matrix you could change the assignment of `fm, features` to `_, features` since we are just throwing away the feature matrix data in that case.

Reply:
good call. Added checks for the null values in the feature matrix, and it means now we're not using the `features` list, so changed that to `_`.

Also moved the test to `test_transform_features`, because it seems that that's the better location for how a specific primitive's feature ends up looking in the feature matrix